Source/DafnyLanguageServer/Workspace/Notifications/VerificationDiagnosticsParams.cs

```diff
@@ -352,54 +366,82 @@ Range Range
   public record DocumentVerificationTree(
     string Identifier,
     int Lines
-  ) : VerificationTree("Document", Identifier, Identifier,
-    new Range(new Position(0, 0),
-      new Position(Lines, 0)));
+  ) : VerificationTree("Document", Identifier, Identifier, Identifier,
+    LinesToRange(Lines)) {
+
+    public static Range LinesToRange(int lines) {
+      return new Range(new Position(0, 0),
+        new Position(lines, 0));
+    }
+  }
 
   public record TopLevelDeclMemberVerificationTree(
+    string Kind,
     string DisplayName,
     // Used to re-trigger the verification of some diagnostics.
     string Identifier,
     string Filename,
     // The range of this node.
     Range Range
-  ) : VerificationTree(DisplayName, Identifier, Filename, Range) {
+  ) : VerificationTree(Kind, DisplayName, Identifier, Filename, Range) {
     // Recomputed from the children which are ImplementationVerificationTree
-    public List<AssertionBatchVerificationTree> AssertionBatches { get; set; } = new();
+    public ImmutableDictionary<(int, int), AssertionBatchVerificationTree> AssertionBatches { get; private set; } =
```

Comment:
Could you replace the anonymous tuple in this field with a record to make it easier to understand what the field contains?

Reply:
I replaced it with the record `ASsertionBatchIndex` for now